### DISCUSSION TITLE
Hide Column but keep Filter

### DISCUSSION THREAD

**Discussion by bitttttten** [TanStack/table#3398] [Q&A]
I have a use case where I want to hide the column, but keep the filter.

In the following sandbox, you can hide and show the filters (like "age" or "first name") and see the filters disappear too: https://codesandbox.io/s/strange-northcutt-mydce?file=/src/App.js

Is there an option to hide the column but keep the filter? So I still want to hide the age column, but let people filter by age still.

**Comment by bruno-fialho** [comment]
Yes, there is.. add the column to initialState > hiddenColumns at useTable props, and you still can setFilter to this column:
```ts
const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    rows,
    prepareRow,
    setFilter,
  } = useTable<ColumnsProps>(
    {
      columns,
      data,
      initialState: {
        hiddenColumns: ['ACTIVE'],
        filters: [{ id: 'ACTIVE', value: 1 }],
      },
    },
    useFilters,
    useSortBy,
  );
```
